### ISSUE TITLE
JSON parsing should be refactored

### ISSUE THREAD

**Issue by lvh** [rackerlabs/mimic#226]
Maybe a decorator or something?

```python
        try:
            content = loads(request.content.read())
        except ValueError:
            request.setResponseCode(400)
            return dumps(bad_request("Invalid JSON request body"))
```

**Comment by glyph** [comment]
Yes, definitely. Possibly this decorator should be in Klein, not in Mimic.

**Comment by cyli** [comment]
The actual error response JSON may differ from service to service.  And in fact, differs inside Nova itself - "Invalid JSON request body" is for creating a Nova server, but if you try to set metadata and the JSON is wrong, the response message is "Malformed request body".

So possibly whatever decorator this is should take a callable that sets the body and status code, if it fails to parse?

**Comment by glyph** [comment]
> So possibly whatever decorator this is should take a callable that sets the body and status code, if it fails to parse?

It does seem like that would be a hard requirement; we should also be able to customize them separately though, because 400 is the right status code 99% of the time, but the error message changes around a fair amount.
